packages/transport/src/modules/relationships/RelationshipsController.ts

```diff
@@ -168,6 +168,35 @@ export class RelationshipsController extends TransportController {
             if (result.error.code === "error.platform.validation.relationship.peerIsToBeDeleted") {
                 throw CoreErrors.relationships.activeIdentityDeletionProcessOfOwnerOfRelationshipTemplate();
             }
+
+            if (result.error.code === "error.platform.validation.relationshipRequest.relationshipToTargetAlreadyExists") {
+                const queryForExistingRelationship = {
+                    "peer.address": template.cache.createdBy,
+                    $or: [
+                        {
+                            ["status"]: { $eq: RelationshipStatus.Pending }
+                        },
+                        {
+                            ["status"]: { $eq: RelationshipStatus.Active }
+                        },
+                        {
+                            ["status"]: { $eq: RelationshipStatus.Terminated }
+                        },
+                        {
+                            ["status"]: { $eq: RelationshipStatus.DeletionProposed }
+                        }
+                    ]
+                };
+
+                const existingRelationshipToPeer = await this.getRelationships(queryForExistingRelationship);
+
+                if (existingRelationshipToPeer.length !== 0) {
+                    throw CoreErrors.relationships.alreadyExists(existingRelationshipToPeer[0].status);
+                }
```

Comment:
you should do that **before** calling the backbone with this.client.createRelationship.

Reply:
Ahh I see, because information from the Backbone is only required if I throw the `CoreErrors.relationships.notYetDecomposedByPeer`?